python/triton/language/semantic.py

```diff
@@ -250,6 +250,59 @@ def mod(input: tl.tensor, other: tl.tensor, builder: ir.builder) -> tl.tensor:
     assert False
 
 
+##############
+# other arithmetic ops
+##############
+
+
+def minimum(arg0: tl.tensor, arg1: tl.tensor, propagate_nan: tl.PropagateNan, builder: ir.builder):
+    arg0, arg1 = binary_op_type_checking_impl(arg0, arg1, builder)
+    dtype = arg0.dtype
+
+    if dtype.is_floating():
+        if propagate_nan == tl.PropagateNan.ALL:
+            if builder.options.has_native_nan_propagation:
+                return tl.tensor(builder.create_minimumf(arg0.handle, arg1.handle), arg0.type)
+            else:
+                r = tl.tensor(builder.create_minnumf(arg0.handle, arg1.handle), arg0.type)
+                nans = tl.isnan(arg0, _builder=builder).__or__(tl.isnan(arg1, _builder=builder), _builder=builder)
+                return where(nans, tl.nan(dtype, _builder=builder), r, _builder=builder)
```

Comment:
wouldn't this be handled by LLVM backend? Or handled by the convert ttgir to llvm dialect? It doesn't feel very intuitive to handle this in the front end.

Reply:
LLVM seems to lower `arith.minimumf` to min.NaN regardless of anything.
I agree that handling this in the frontend is a bit controversial. It was proposed, because it was needed only for V100, so there may be a value in keeping MLIR part simpler. But I am open to suggestions.

Reply:
Maybe we should deprecate V100 support already, actually. I have heard that there are some bugs related to chained matmuls and transpositions (e.g., flash attention bwd pass) and it is clearly not worth the effort to try and fix them. How about we officially stop supporting the architecture and let users do the workaround in their kernel?

Reply:
makes sense, let's not fix it for now then, it is not worth the extra complexity in the front end